app/Models/User.php

```diff
@@ -363,10 +363,12 @@ public static function lookup($username_or_id, $lookup_type = null, $find_all =
 
             default:
                 if (is_numeric($username_or_id)) {
-                    $user = self::where('user_id', $username_or_id);
-                } else {
-                    $user = self::where('username', $username_or_id)->orWhere('username_clean', '=', $username_or_id);
+                    $user = self::lookup($username_or_id, 'id', $find_all);
+                    if ($user !== null) {
+                        return $user;
+                    }
                 }
+                $user = self::where('username', $username_or_id)->orWhere('username_clean', '=', $username_or_id);
```

Comment:
```
default:
  if (is_numeric) {
    $user = ...;
  }
  return $user ?? static::lookup(...);
```

Reply:
aha I knew there had to be a better way. thanks